src/stdune/path.ml

```diff
@@ -1335,3 +1335,16 @@ let chmod ~mode ?(stats = None) ?(op = `Set) path =
         stats.st_perm land lnot mode
   in
   Unix.chmod (to_string path) mode
+
+let realpath t =
+  let path_s = to_absolute_filename t in
+  let stat = Unix.lstat path_s in
+  match stat.st_kind with
+  | S_LNK ->
+    Some
+      (let link = Unix.readlink path_s in
+       if Filename.is_relative link then
+         relative (parent_exn t) link
```

Comment:
If symlink points outside of the build directory, then we crash here, which seems undesirable. Maybe the whole function should work with simple unix paths instead of with `Path.t`?

Reply:
Good idea. Moved it to Fpath.